### DISCUSSION TITLE
get inactive users

### DISCUSSION THREAD

**Discussion by botkero** [python-gitlab/python-gitlab#2300] [Q&A] [answered]
hello,

how do I get users who have been inactive since x months or so.
I found this in the API-doc: https://python-gitlab.readthedocs.io/en/stable/gl_objects/users.html#users-activities

```py
import gitlab

gl = gitlab.Gitlab(url=url, private_token=token)
r = gl.user_activities.list(query_parameters={'from': '2022-10-07'}, get_all=True) 
```

**Comment by nejch** [comment, marked as answer]
@botkero I thought you had this figured out already but I see you've edited your question.

Just to make sure for future readers, your example should work, provided that you have an admin token. The query_parameters are needed due to the collision with python's `from` keyword and the date format is slightly different from the ISO format usually used, as shown in the upstream link above.

**Reply by botkero** [reply]
yea sorry, I made too many mistakes and reduced my post to the essentials. And tried a few things more. 

As far as I can see, the "from" give me users that where active since the given period. But I want the opposite (I want users who have not logged in for a specified period of time). Is there any other argument I can use instead of 'from'?

**Reply by nejch** [reply]
Ah I misread your initial post then, sorry. GitLab's API does not offer that, no. However, if you only care about "users who have not logged in" (which is technically different from last activity, I think) you can achieve this by iterating over all users and filtering that based on their `last_sign_in_at` value.

Quick mock-up:

```python
def is_inactive(user):
    # some logic that parses the value of `user.last_activity_at`

inactive_users = [user for user in gl.users.list(iterator=True) if is_inactive(user)]
```

Since you need administrator access for the original question, I assume you're on a self-hosted instance so iterating over all users should still be feasible (I guess it'll be at most a few ten thousand users for large self-hosted)
